dataprofiler/profilers/profile_builder.py

```diff
@@ -2112,6 +2130,161 @@ def _update_chi2(self):
 
         return chi2_mat
 
+    def _get_null_and_sample_count(self):
+        """
+        Gets the null_count and sample_size of each column.
+
+        :return: Dictionary where column id maps to a list of [null_count, sample_size]
+        :rtype: dict
+        """
+        counts = {}
+        for col_id in range(len(self._profile)):
+            profile = self._profile[col_id]
+            null_count = getattr(profile, "null_count", 0)
+            sample_size = getattr(profile, "sample_size", 0)
+            counts[col_id] = [null_count, sample_size]
+        return counts
+
+    def _calc_null_replication_metrics(self, clean_samples, prev_counts):
+        """
+        Calculate metrics needed for replicating null values in capitalone/synthetic-data.
+
+        Metrics include class priors and class specific means of each column.
+
+        Required for running LDA based binary classifier where predicted class label indicates whether
+        a value of a column should be NaN (1) or not (0).
+
+        :param clean_samples: input cleaned dataset
+        :type clean_samples: dict
+        :param prev_counts: null_count and sample_size of each column before update
+        :type prev_counts: dict
+        """
+        data = pd.DataFrame(clean_samples).apply(pd.to_numeric, errors="coerce")
+
+        for col_id in range(len(self._profile)):
+            profile = self._profile[col_id]
+            null_count = getattr(profile, "null_count")
+            if null_count == 0:
+                # No missing values to replicate
+                continue
+
+            # Calculate class priors
+            # i.e probability that a value in target col is or is not NaN
+            sample_size = getattr(profile, "sample_size")
+            true_count = sample_size - null_count
+
+            # null_count and sample_size get updated on profile_update
+            # Therefore the priors are calculated from the updated values
+            prior_null = null_count / sample_size
+            prior_not_null = true_count / sample_size
+
+            # Partition data based on whether target column value is null or not
+            col_is_null = data.iloc[:, col_id].isnull()
+            partition_null = data[col_is_null]
+            partition_not_null = data[~col_is_null]
+
+            # Drop the target column from partitions
+            partition_null = partition_null.drop(columns=col_id)
+            partition_not_null = partition_not_null.drop(columns=col_id)
```

Comment:
Can we index the values with the boolean without dropping? Calculate the means without dropping?

Reply:
The only reason we drop the `col_id` (which is the column where we will be replicating null values in the synthetic data) is because we do not want to have the mean of that column in the mean vector. So we could calculate the mean first and then drop the column but that would be more expensive (calculating the mean of a column we don't need)

Reply:
I forget though if slicing also copies though.